lightning/src/routing/scoring.rs

```diff
@@ -771,6 +771,19 @@ impl ProbabilisticScoringDecayParameters {
 	}
 }
 
+/// A dummy copy of [`ChannelLiquidity`] to calculate its unpadded size
+#[repr(C)]
+struct DummyLiquidity {
+	a: u64,
+	b: u64,
+	c: HistoricalLiquidityTracker,
+	d: Duration,
+	e: Duration,
+}
+
+/// The amount of padding required to make [`ChannelLiquidity`] (plus a u64) a full 4 cache lines.
+const LIQ_PADDING_LEN: usize = (256 - ::core::mem::size_of::<(u64, DummyLiquidity)>()) / 8;
```

Comment:
Do you have any idea on the effect size of this optimization?

Reply:
Not sure if it is worth it then, writing, reviewing, understanding and maintaining the code and making sure that it remains a win on all platforms.

Reply:
Linear scoring - what does this mean exactly? That liquidity information is retrieved for channels in the order in which they are stored in the hashmap? Interested to hear where 'linear scoring' is happening in LDK.

Reply:
You are saying that when linear scoring is removed, non-linear floating point conversion takes over, and this cache optimization isn't significant anymore. Is it worth keeping then?